server-spi-private/src/main/java/org/keycloak/broker/provider/BrokeredIdentityContext.java

```diff
@@ -48,6 +48,7 @@ public class BrokeredIdentityContext {
     private IdentityProviderModel idpConfig;
     private IdentityProvider idp;
     private Map<String, Object> contextData = new HashMap<>();
+    private Map<String, String> sessionNotes = new HashMap<>();
```

Comment:
Perhaps instead of exposing this map directly to mappers as we are doing here, we should have a method `setUserSessionNote` (similar to what we have in `AuthenticationSession`) and use this method to encapsulate the logic to actually set the session note to later push it into the user session.

This method would basically run the same check you are introducing to the mappers and set the note either to this new map or to the authentication session. Mappers would just call this method and not deal directly with the differences between the different contexts they can run: frontend/broker-login and backend/token-exchange.